workflow/main.py

```diff
@@ -0,0 +1,213 @@
+from json.decoder import JSONDecodeError
+import os
+import pandas as pd
+import numpy as np
+from datetime import date
+import subprocess
+import sys
+import time
+import csv
+
+
+def main():
+    '''
+    Outine:
+        - run TPS
+        - gather input data
+        - prepare annotations file
+        - run visualization
+
+    '''
+
+    today = date.today()
+
+    # * Run TPS!
+
+    # * create Base directory
+    baseDir = os.path.abspath(os.path.join('..'))
+    tps_in_dir = os.path.join(baseDir, 'data', 'timeseries')
+    tps_out_dir = os.path.join(
+        baseDir, 'results', 'results_reproduce_egfr_tps_' + str(today))
+    if not os.path.exists(tps_out_dir):
+        os.makedirs(tps_out_dir)
+        print('Created {}'.format(tps_out_dir))
+    print(baseDir)
+    print(tps_in_dir)
+    print(tps_out_dir)
+
+    # * Move files into directory
+    output_sif_orig = os.path.join(baseDir, 'output.sif')
+    activity_tsv_orig = os.path.join(baseDir, 'activity-windows.tsv')
+    output_dest = os.path.join(baseDir, 'workflow', 'output.sif')
+    activity_dest = os.path.join(baseDir, 'workflow', 'activity-windows.tsv')
+
+    os.replace(output_sif_orig, output_dest)
+    os.replace(activity_tsv_orig, activity_dest)
+
+    # input args from command line
+    OUTPUT_FILE = os.path.join(baseDir, 'workflow', 'output.sif')
+    STYLE_FILE = os.path.join(baseDir, 'workflow', 'tps_style.xml')
+
+    # dir name set manually
+    DIRNAME = r"C:\Users\ajshe\OneDrive\Documents\Comp_bio\Cytoscape_v3.7.1"
+    CYTOSCAPE = 'Cytoscape.exe'
+
+    if OUTPUT_FILE.lower().endswith(".sif"):
+        print("file has valid extention")
+    else:
+        print("invalid extension on output file: ", OUTPUT_FILE)
+        sys.exit()
+
+    # get absolute paths if nessesary
+    OUTPUT_FILE = os.path.abspath(OUTPUT_FILE)
+    STYLE_FILE = os.path.abspath(STYLE_FILE)
+    print("---absolute output path: ", OUTPUT_FILE)
+    print("---absolute style path: ", STYLE_FILE)
+
+    # * Gather data used to generate node anotations
+    # Use the version with the header line
+    pepMapFile = os.path.join(tps_in_dir, 'peptide-mapping.tsv')
+    pepFirstFile = os.path.join(tps_in_dir, 'p-values-first.tsv')
+    pepPrevFile = os.path.join(tps_in_dir, 'p-values-prev.tsv')
+    print(pepMapFile)
+    print(pepFirstFile)
+    print(pepPrevFile)
+
+    # Use the version for which log2 fold change has been precomputed
+    timeSeriesFile = os.path.join(
+        baseDir, 'data', 'timeseries', 'log2FoldChange011215.txt')
+    #timeSeriesFile = r"C:\\Users\\ajshe\\Anaconda3\\envs\\py2cyto\\tps\\data\\timeseries\\log2FoldChange011215.txt"
+    print(timeSeriesFile)
+
+    windowsFile = os.path.join(baseDir, 'workflow', 'activity-windows.tsv')
+    networkFile = os.path.join(baseDir, 'workflow', 'output.sif')
+    print(windowsFile)
+    print(networkFile)
+
+    # Use the same EGFR gold standard for Olsen 2006 and Ale's data
+    goldStandardFile = os.path.join(
```

Comment:
This type of file may not always be available.  Will the script work if it is not present?

Reply:
From what I have tested, the script does not work if this file is completely omitted. We can add a check in the `table_generation.py`